resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshotsStoreImpl.scala

```diff
@@ -57,6 +60,7 @@ import org.apache.spark.util.ThreadUtils
  * The subscriber notification callback is guaranteed to be called from a single thread at a time.
  */
 private[spark] class ExecutorPodsSnapshotsStoreImpl(
+    conf: SparkConf,
```

Comment:
Although this is a `private`, please provide a backward-compatibility in this case by keeping the existing constructor and adding a new one with new parameter `conf`.

Reply:
the backward-compatible constructors were added

Reply:
If `conf` is placed after `clock` and with a default value `newSparkConf` will there be compatibility problems? If not, we can delete the other two constructors

Reply:
I moved `conf: SparkConf = new SparkConf` to the last as suggested